examples/contrib/detection/faster_rcnn.py

```diff
@@ -0,0 +1,286 @@
+
+        if device == "cuda":
+            with torch.cuda.amp.autocast(enabled=True):
+                loss_dict = model(images, targets)
+                loss = sum(loss for loss in loss_dict.values())
+
+            optimizer.zero_grad()
+            scaler.scale(loss).backward()
+            scaler.step(optimizer)
+            scaler.update()
+        else:
+            loss_dict = model(images, targets)
+            loss = sum(loss for loss in loss_dict.values())
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+        loss_items = {k: v.item() for k, v in loss_dict.items()}
+        loss_items["loss_average"] = loss.item() / 4
+
+        return loss_items
+
+    @torch.no_grad()
+    def inference(engine, batch):
+        model.eval()
+        images, targets = batch
+        images = list(image.to(device) for image in images)
+        outputs = model(images)
+        outputs = [{k: v.to("cpu") for k, v in t.items()} for t in outputs]
+        return {"y_pred": outputs, "y": targets, "x": [i.cpu() for i in images]}
+
+    trainer = Engine(update_model)
+    evaluator = Engine(inference)
+    visualizer = Engine(inference)
+
+    aim_logger = AimLogger(
+        repo=os.path.join(log_dir, "aim"),
+        experiment=experiment_name,
+    )
+
+    CocoMetric(convert_to_coco_api(val_dataset)).attach(evaluator, "mAP")
+
+    @trainer.on(Events.EPOCH_COMPLETED)
+    def log_validation_results(engine):
+        evaluator.run(val_dataloader)
+        visualizer.run(vis_dataloader)
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def step_scheduler(engine):
+        scheduler.step()
+        aim_logger.log_metrics({"lr": scheduler.get_last_lr()[0]}, step=engine.state.iteration)
+
+    @visualizer.on(Events.EPOCH_STARTED)
+    def reset_vis_images(engine):
+        engine.state.model_outputs = []
+
+    @visualizer.on(Events.ITERATION_COMPLETED)
+    def add_vis_images(engine):
+        engine.state.model_outputs.append(engine.state.output)
+
+    @visualizer.on(Events.ITERATION_COMPLETED)
+    def submit_vis_images(engine):
+        aim_images = []
+        for outputs in engine.state.model_outputs:
+            for image, target, pred in zip(outputs["x"], outputs["y"], outputs["y_pred"]):
+                image = (image * 255).byte()
+                pred_labels = [Dataset.class2name[l.item()] for l in pred["labels"]]
+                pred_boxes = pred["boxes"].long()
+                image = draw_bounding_boxes(image, pred_boxes, pred_labels, colors="red")
+
+                target_labels = [Dataset.class2name[l.item()] for l in target["labels"]]
+                target_boxes = target["boxes"].long()
+                image = draw_bounding_boxes(image, target_boxes, target_labels, colors="green")
+
+                aim_images.append(aim.Image(image.numpy().transpose((1, 2, 0))))
+        aim_logger.experiment.track(aim_images, name="vis", step=trainer.state.epoch)
```

Comment:
I'm not quite sure to understand why we need to accumulate and submit on iterations...
Can't we just submit on iterations without accumulating ?

Reply:
Can submit on iterations only if we set `step` as `trainer.state.iteration`, or `aim` will override the previous images.

Gather visualization images in epoch can give a good view when using aim I think, but indeed it will eat more memory.